src/spark_manager.py

```diff
@@ -354,3 +357,83 @@ def _create_or_replace_deployment(
             else:
                 logger.error(f"Error creating {resource_description}: {e}")
                 raise
+
+    def get_cluster_status(self) -> SparkClusterStatus:
+        """
+        Get the status of the Spark cluster for the authenticated user.
+
+        Returns:
+            SparkClusterStatus: The current status of the Spark cluster
+        """
+        master_status = self._get_deployment_status(self.master_name)
+        worker_status = self._get_deployment_status(self.worker_name)
+
+        master_url = None
+        master_ui_url = None
+
+        # Add the master URL if the master has ready replicas
+        if master_status.ready_replicas > 0:
+            master_url = f"spark://{self.master_name}.{self.namespace}:{self.DEFAULT_MASTER_PORT}"
+            master_ui_url = f"http://{self.master_name}.{self.namespace}:{self.DEFAULT_MASTER_WEBUI_PORT}"
+
+        status = SparkClusterStatus(
+            master=master_status,
+            workers=worker_status,
+            master_url=master_url,
+            master_ui_url=master_ui_url,
+        )
+
+        return status
+
+    def _get_deployment_status(self, deployment_name: str) -> DeploymentStatus:
+        """
+        Get the current status of a deployment.
+        """
+
+        status = DeploymentStatus(
+            available_replicas=0,
+            ready_replicas=0,
+            replicas=0,
+            unavailable_replicas=0,
+            is_ready=False,
+            exists=False,
```

Comment:
The only one of these that isn't the default is exists, why set the others?

Reply:
👍 removed.

Reply:
:+1: